[Doc] Move actor checkpointing to actor fault tolerance page (#32153)
ray-project/ray: doc/source/ray-core/fault_tolerance/actors.rst

upon actor restart.

Actor checkpointing
~~~~~~~~~~~~~~~~~~~

> Although ``max_restarts`` will restart the crashed actor automatically,

Reviewer: ```suggestion
``max_restarts`` automatically restarts the crashed actor,
```

> but it doesn't allow you to restore application level state in your actor.

Reviewer 2: ```suggestion
but it doesn't automatically restore application level state in your actor.
```

> it doesn't provide ways for you to restore application level state in your actor.

Reviewer: ```suggestion
but it doesn't allow you to restore application level state in your actor.
```

> You can checkpoint your actor's state and recover from the checkpoint with an actor restart.

Reviewer 2: ```suggestion
Instead, you should manually checkpoint your actor's state and recover upon actor restart.
```

> For this, you can checkpoint your actor's state and recover from the checkpoint upon actor restart.

Reviewer: ```suggestion
You can checkpoint your actor's state and recover from the checkpoint with an actor restart.
```

> Let the actor creator manage the checkpoint and manually restart and recover the actor upon failure:

Reviewer 2: ```suggestion
For actors that are restarted manually, the actor's creator should manage the checkpoint and manually restart and recover the actor upon failure. This is recommended if you want the creator to decide when the actor should be restarted and/or if the creator is coordinating actor checkpoints with other execution:
```

> One way of doing this is letting the actor creator manage the checkpoint and manually restart and recover the actor upon failure:

Reviewer: ```suggestion
Let the actor creator manage the checkpoint and manually restart and recover the actor upon failure:
```

.. literalinclude:: ../doc_code/actor_checkpointing.py
  :language: python
  :start-after: __actor_checkpointing_manual_restart_begin__
  :end-before: __actor_checkpointing_manual_restart_end__

> Alternatively, the actor can checkpoint and restore automatically:

Reviewer 2: ```suggestion
Alternatively, if you are using Ray's automatic actor restart, the actor can checkpoint itself manually and restore from a checkpoint in the constructor:
```

> Or the actor can checkpoint and restore automatically:

Reviewer: ```suggestion
Alternatively, the actor can checkpoint and restore automatically:
```

.. literalinclude:: ../doc_code/actor_checkpointing.py
  :language: python
  :start-after: __actor_checkpointing_auto_restart_begin__
  :end-before: __actor_checkpointing_auto_restart_end__

.. note::

>   If the checkpoint is saved to an external storage, you need to make sure

Reviewer: ```suggestion
  If the checkpoint is saved to external storage, make sure
```

>   it's accessible to the entire cluster since the actor might be restarted

Reviewer: ```suggestion
  it's accessible to the entire cluster, since the actor might be restarted
```

  on a different node.

>   Save the checkpoint to cloud storage (e.g., S3) or a shared directory (e.g., via NFS).

Reviewer 2: ```suggestion
  For example, save the checkpoint to cloud storage (e.g., S3) or a shared directory (e.g., via NFS).
```

>   You can save the checkpoint to a cloud storage (e.g. S3) or a shared directory (e.g. via NFS).

Reviewer: ```suggestion
  Save the checkpoint to cloud storage (e.g., S3) or a shared directory (e.g., via NFS).
```
